### DISCUSSION TITLE
Converting pixi graphics to a texture

### DISCUSSION THREAD

**Discussion by Nukiloco** [pixijs/pixijs#8304] [General]
What is a way to convert a PIXI graphics to a regular texture without having to convert it to a render texture?

**Comment by bigtimebuddy** [comment]
Couple options:
* Render your Graphics to a separate CanvasRenderer and use the canvas element as the Texture resource
* Use the Renderer extract plugin to create an HTMLImage element, canvas, ImageData or base64 string, all could be basis for a Texture element

**Reply by Nukiloco** [reply]
Where would the Renderer extract plugin be exactly located at? I can't seem to find this plugin at all.

**Reply by Nukiloco** [reply]
Also how would someone go about using a separate canvas render? Is there any example of this?

**Reply by bigtimebuddy** [reply]
The [Extract plugin](http://pixijs.download/dev/docs/PIXI.Extract.html) is built-in to Pixi. To use it you can do something like this:
```js
renderer.plugins.extract.canvas(graphics)
```

To create a new CanvasRenderer, you can do this:

```js
// Create a canvas renderer
const renderer = new CanvasRenderer();
renderer.addChild(graphics);
renderer.render();
// Create the new texture
Texture.from(renderer.view);
```

These aren't full examples. Please search the issues and discussion and you should be able to find a bunch of references to CanvasRenderer or Extract. There's also [examples](https://pixijs.io/examples) and [guides](https://pixijs.io/guides) if you're looking for more general information or examples.
